src/overlays/actors/ovl_player_actor/z_player.c

```diff
@@ -8641,11 +8652,19 @@ void Player_Action_80841BA8(Player* this, PlayState* play) {
 
     Player_GetMovementSpeedAndYaw(this, &speedTarget, &yawTarget, SPEED_MODE_CURVED, play);
 
-    if (!Player_TryActionHandlerList(play, this, sActionHandlerList6, true)) {
+    //! @bug This action does not handle xzSpeed in any capacity.
+    //! Player's current speed value will be maintained the entire time this action is running.
+    //! This is the core bug that allows many different glitches to manifest.
+    //!
+    //! One possible fix is to kill all speed instantly in `Player_SetupTurnInPlace`.
+    //! Another possible fix is to gradually kill speed by calling `Player_DecelerateToZero`
+    //! here, which plenty of other "standing" actions do.
+
+    if (!Player_TryActionHandlerList(play, this, sActionHandlerListTurnInPlace, true)) {
         if (speedTarget != 0.0f) {
             this->actor.shape.rot.y = yawTarget;
             func_8083C858(this, play);
-        } else if (Math_ScaledStepToS(&this->actor.shape.rot.y, yawTarget, this->unk_87E)) {
+        } else if (Math_ScaledStepToS(&this->actor.shape.rot.y, yawTarget, this->turnRate)) {
             func_8083C0E8(this, play);
         }
```

Comment:
I know this function is buggy thanks to the xzSpeed not being handled in any way, and it's clearly related to turning, but which part of this function is suppose to be "in place"? I believe `func_8083C858` is suppose to setup running in some way, but would it make sense to call that function to transition to running if this function intended to reduce the xzSpeed to 0?

Reply:
it does still make sense to call `func_8083C858` if you hold the stick far enough, yes. Cause you will have left "ess position" (for lack of a better term), and its time to start moving now.

Its in place because he is (supposed to) stay standing still and shuffle his feet to turn toward the desired angle. VS holding the stick full magnitude in some direction, which would not use this function at all to turn/change direction

Reply:
Additionally, this action only runs when `speedTarget` is zero when being called from other actions. Its only supposed to run when there is 0 speed, so "in place"

Reply:
Yeah, that makes sense. Thanks for clarifying